This patch works around an issue with vim dumping core with SIGSEGV when
handling a more benign signal (typically SIGHUP or SIGTERM). Vim tries to
run all the exit-vim autocommands, which end up busting the alternate stack
at its default size. Increasing the default seems to do the trick, but it
continues to be just a guess.

The second change (already in upstream) comes from Bram:
https://groups.google.com/d/msg/vim_dev/GrLf2ePQWDk/Gf3AMnjIAQAJ

diff --git a/src/os_unix.c b/src/os_unix.c
index 01f5f59..6069232 100644
--- a/src/os_unix.c
+++ b/src/os_unix.c
@@ -774,9 +774,10 @@ mch_stackcheck(char *p)
  * completely full.
  */
 
-#ifndef SIGSTKSZ
-# define SIGSTKSZ 8000    /* just a guess of how much stack is needed... */
+#ifdef SIGSTKSZ
+# undef SIGSTKSZ
 #endif
+#define SIGSTKSZ 8*1024*1024  /* just a guess of how much stack is needed... */
 
 # ifdef HAVE_SIGALTSTACK
 static stack_t sigstk;			/* for sigaltstack() */
